corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -233,9 +242,147 @@ hqDefine("geospatial/js/case_grouping_map",[
+                if (error) {
+                    reject(error);
+                } else {
+                    resolve(casePoints);
+                }
+            });
+        });
+    }
+
+    function setMapLayersVisibility(visibility) {
+        map.setLayoutProperty('clusters', 'visibility', visibility);
+        map.setLayoutProperty('cluster-count', 'visibility', visibility);
+        map.setLayoutProperty('unclustered-point', 'visibility', visibility);
+    }
+
+    function getRandomColor() {
+        const randomColor = Math.floor(Math.random() * 16777215).toString(16);
+        return `#${randomColor}`;
+    }
+
+    function collapseGroupsOnMap() {
+        setMapLayersVisibility(MAPBOX_LAYER_VISIBILITY.None);
+        var groupColors = {};
+
+        exportModelInstance.casesToExport().forEach(function (caseItem) {
+            const groupId = caseItem.groupId;
+            if (groupId) {
+                if (groupColors[groupId] === undefined) {
+                    groupColors[groupId] = getRandomColor();
+                }
+                const color = groupColors[groupId];
+
+                const marker = new mapboxgl.Marker({ color: color, draggable: false });  // eslint-disable-line no-undef
+                marker.setLngLat([caseItem.coordinates.lng, caseItem.coordinates.lat]);
+
+                // Add the marker to the map
+                marker.addTo(map);
+                mapMarkers.push(marker);
+            }
+        });
+    }
+
+    function uuidv4() {
+        // https://stackoverflow.com/questions/105034/how-do-i-create-a-guid-uuid/2117523#2117523
+        return "10000000-1000-4000-8000-100000000000".replace(/[018]/g, c =>
+            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16),
+        );
+    }
+
+    async function setCaseGroups() {
+        const sourceFeatures = map.querySourceFeatures('caseWithGPS', {
+            sourceLayer: 'clusters',
+            filter: ['==', 'cluster', true],
+        });
+        const clusterSource = map.getSource('caseWithGPS');
+        let caseGroups = {};
+
+        for (const cluster of sourceFeatures) {
+            const clusterId = cluster.properties.cluster_id;
+            const pointCount = cluster.properties.point_count;
+
+            try {
+                const casePoints = await getClusterLeavesAsync(clusterSource, clusterId, pointCount);
+                const groupUUID = uuidv4();
+                for (const casePoint of casePoints) {
+                    const caseId = casePoint.properties.id;
+                    caseGroups[caseId] = {
+                        groupId: groupUUID,
+                        groupCoordinates: {
+                            lng: cluster.geometry.coordinates[0],
+                            lat: cluster.geometry.coordinates[1],
+                        },
+                    };
+                }
+            } catch (error) {
+                console.error("Error processing cluster:", error);
```

Comment:
Was this log left in intentionally?

Reply:
Kinda, I'm not sure if I should leave this or report an issue to the user (maybe through an alert?). Any thoughts?

Reply:
I think it would be good to have an UI alert that something has gone wrong for the user. Console logs aren't normally visible to end-users, and so they might not know that an issue has occurred. This will also help ensure that users aren't under the assumption that the export contains all their cases, when in fact there were hidden errors and some clusters were never processed.